crates/validation_common/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+mod digest;
+
+pub use digest::Digest;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum Measurements {
+    /// Only Enarx signer information is available. This is the key which
+    /// has signed the Enarx binary, directly or indirectly.
+    Signer { enarx_signer: Vec<Digest> },
+
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    Hash { enarx_hash: Vec<Digest> },
+
+    /// Both the Enarx hash and hash of a signing key are present.
+    Both {
+        enarx_hash: Vec<Digest>,
+        enarx_signer: Vec<Digest>,
+    },
+}
+
+impl Default for Measurements {
+    fn default() -> Self {
+        Measurements::Both {
+            enarx_hash: Default::default(),
+            enarx_signer: Default::default(),
+        }
+    }
+}
```

Comment:
I don't think this is valid, this is not "both" - this is "none".
We explicitly wanted to ensure that at least one of these fields is set, which is why this enum exists in the first place, but this is an escape hatch, which allows for not specifying either.

Reply:
Fixed by your diff